File: module-common/src/main/java/com/geekhome/common/configurable/RequiredLongValidator.java
package com.geekhome.common.configurable;

import com.geekhome.common.localization.Resource;

public class RequiredLongValidator implements Validator<Long> {
    @Override
    public Resource getReason() {
        return new Resource("This field is required",
                "To pole jest wymagane");
    }

    @Override
    public boolean validate(Long fieldValue) {
        return fieldValue != null && fieldValue > 0;
    }
}

File: module-domain/src/main/java/eu/geekhome/domain/configurable/RequiredLongValidator.kt
package eu.geekhome.domain.configurable

import eu.geekhome.domain.R
import eu.geekhome.data.localization.Resource

class RequiredLongValidator : Validator<Long> {

    override fun validate(fieldValue: Long?, fields: Map<String, String?>): Boolean {
        if (fieldValue == null) {
            return false
        }

        return fieldValue > 0
    }

    override val reason: Resource = R.validator_required_field
}